18F/identity-idp
# LG-13402: Prompt user to confirm setting up backup codes from account page

## 🎫 Ticket

[LG-13402](https://cm-jira.usa.gov/browse/LG-13402)

## 🛠 Summary of changes

Reintroduces the confirmation screen for setting up backup codes, when adding backup codes through the account page.

In LG-12030, we removed the screen which was shown before setting up backup codes confirming whether the user wanted to add backup codes to their account. This was removed primarily for account creation, since the user would be presented with several screens before and after the actual creation of backup codes to warn them about their potential risks, and it was overwhelming. While this remains true for account creation, adding backup codes to an existing account does not have the same set of screens occurring after the creation of backup codes.

Additionally, a user may simply be exploring navigation items available for their account, and it would be surprising that merely clicking on a navigation item results in the creation of backup codes that they may or may not want.

## 📜 Testing Plan

Verify that account creation is unimpacted, by repeating the Testing Plan from #10014

Verify that you are prompted to confirm adding backup codes when added through the account dashboard:

1. Go to http://localhost:3000
2. Sign in
3. From account dashboard, click "Get backup codes" in the sidebar
4. Observe that you are shown the screen in "Screenshots" below
5. Click "Continue" or "Cancel"
6. Observe:
   1. If you click "Continue", you are presented with your backup codes
   2. If you click "Cancel", you are returned to the preceding screen and have not added backup codes to your account

## 👀 Screenshots

![image](https://github.com/18F/identity-idp/assets/1779930/80cb7139-c6ef-4f15-a398-9dd7783a7c32)
